src/CoreWCF.Primitives/src/CoreWCF/Runtime/SynchronizedPool.cs

```diff
@@ -10,40 +10,40 @@ namespace CoreWCF.Runtime
+    // which takes locks.  As different threads take and return values, we record their IDs,
+    // and if we detect that a thread is taking and returning "enough" on the same thread,
+    // then we decide to "promote" the thread.  When a thread is promoted, we decrease the
+    // quota of the global pool by one, and allocate a thread-specific entry for the thread
+    // to store it's value.  Once this entry is allocated, the thread can take and return
+    // it's value from that entry without taking any locks.  Not only does this avoid
     // locks, but it affinitizes pooled items to a particular thread.
     //
     // There are a couple of additional things worth noting:
-    // 
+    //
     // It is possible for a thread that we have reserved an entry for to exit.  This means
-    // we will still have a entry allocated for it, but the pooled item stored there 
-    // will never be used.  After a while, we could end up with a number of these, and 
+    // we will still have a entry allocated for it, but the pooled item stored there
+    // will never be used.  After a while, we could end up with a number of these, and
     // as a result we would begin to exhaust the quota of the overall pool.  To mitigate this
-    // case, we throw away the entire per-thread pool, and return all the quota back to 
-    // the global pool if we are unable to promote a thread (due to lack of space).  Then 
+    // case, we throw away the entire per-thread pool, and return all the quota back to
+    // the global pool if we are unable to promote a thread (due to lack of space).  Then
     // the set of active threads will be re-promoted as they take and return items.
-    // 
+    //
     // You may notice that the code does not immediately promote a thread, and does not
-    // immediately throw away the entire per-thread pool when it is unable to promote a 
-    // thread.  Instead, it uses counters (based on the number of calls to the pool) 
+    // immediately throw away the entire per-thread pool when it is unable to promote a
+    // thread.  Instead, it uses counters (based on the number of calls to the pool)
     // and a threshold to figure out when to do these operations.  In the case where the
-    // pool to misconfigured to have too few items for the workload, this avoids constant 
+    // pool to misconfigured to have too few items for the workload, this avoids constant
     // promoting and rebuilding of the per thread entries.
     //
     // You may also notice that we do not use interlocked methods when adjusting statistics.
-    // Since the statistics are a heuristic as to how often something is happening, they 
+    // Since the statistics are a heuristic as to how often something is happening, they
     // do not need to be perfect.
-    // 
-    internal class SynchronizedPool<T> where T : class
+    //
+    public class SynchronizedPool<T> where T : class
     {
```

Comment:
Hopefully we can work out how to replace this with ObjectPool. Leaving this comment here so it doesn't get missed in that discussion.

Reply:
As discussed we decided to leave this alone for now.